sql/api/src/main/scala/org/apache/spark/sql/streaming/ValueState.scala

```diff
@@ -42,8 +43,22 @@ private[sql] trait ValueState[S] extends Serializable {
   /** Get the state if it exists as an option and None otherwise */
   def getOption(): Option[S]
 
-  /** Update the value of the state. */
-  def update(newState: S): Unit
+  /**
+   * Update the value of the state.
+   * @param newState the new value
+   * @param ttlDuration set the ttl to current batch processing time
+   *                    (for processing time TTL mode) plus ttlDuration
+   */
+  def update(newState: S, ttlDuration: Duration = Duration.ZERO): Unit
+
+
+  /**
+   * Update the value of the state.
+   *
+   * @param newState    the new value
+   * @param expirationMs set the ttl to expirationMs (processingTime or eventTime)
```

Comment:
nit: should we rename to `expirationTimeInMs` ?

Reply:
done.